content/articles/custom-decorators-in-django/index.md

```diff
@@ -0,0 +1,499 @@
+            if user is not None:
+                auth.login(request, user)
+                messages.success(request, "Logged in Successfully!")
+                print("Logged in Successfully!")
+                return redirect("accounts:profile")
+            else:
+                messages.error(request, "Invalid credentials, wrong username or password")
+                print("Invalid credentials, wrong username or password")
+        else:
+            messages.error(request, "form invalid")
+            print("form invalid")
+    return render(request, 'accounts/login.html', {'form': login_form})
+
+def profile(request):
+    return render(request, 'accounts/profile.html')
+```
+Let’s have a brief overview of what is happening in the above code,
+
+- In line 3, we import auth to use the default authentication function for getting a user object from their username and password.
+- In line 4, we import the LoginForm & RegisterForm from forms.py file to render in the template and users to fill.
+- In the register view, the newly created user has been made inactive intentionally to illustrate the verificaton_required decorator.
+- authentication_not_required decorator has been applied on the register & login view to perform their purpose.
+- In the profile view, we simply render a template named profile.html to the user. The profile.html displays the text profile and the message in the messages framework from our view.
+
+templates/accounts/profile.html
+```html
+{% for message in messages %}
+    <p>{{ message }}</p>
+{% endfor %}
+<h1>Profile</h1>
+```
+templates/accounts/register.html
+```html
+<form action="{% url 'accounts:register' %}" method="post">
+    {% csrf_token %}
+    {{ form.as_p }}
+    <input type="submit" value="Register">
+</form>
+```
+The login.html file is the same as the register.html, just replace register with login.
+Create a new file called `urls.py` to map the route / path to the right view function
+```py
+from django.urls import path
+from .views import (login, profile, register)
+
+app_name = 'accounts'
+urlpatterns = [
+    path('login', login, name="login"),
+    path('register', register, name="register"),
+    path('profile', profile, name="profile"),
+]
+```
+Testing the application written so far…
+Create a superuser and then start your server
+```bash
+python manage.py createsuperuser
+python manage.py runserver
+```
+Go to your browser and navigate to http://127.0.0.1:8000/
+
+Check the image below
+![Authentication Decorators in Action](custom-decorators-in-django/authdec.gif)
+
+
+- A link or an OTP should be sent to the user's email address in order to verify his / her email account and activate the user. I have kept this tutorial simple that the admin should make the user active.
+> The default authentication backend does not return a user object for in active users. So if a user who created an account using the register.html won't be able to login unless the admin makes the user active.
```

Comment:
Replace with...
The default authentication backend does not return a user object for inactive users. So a user who created an account using the 'register.html' won't be able to login unless the admin makes the user active.